provider/builtin/provider.go

```diff
@@ -0,0 +1,116 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/konveyor/analyzer-lsp/jsonrpc2"
+	"github.com/konveyor/analyzer-lsp/provider/lib"
+)
+
+var capabilities = []string{
+	"filecontent",
+	"file",
+	"xml",
+}
+
+type builtinProvider struct {
+	rpc *jsonrpc2.Conn
+	ctx context.Context
+
+	config lib.Config
+}
+
+func NewBuiltinProvider(config lib.Config) *builtinProvider {
+	return &builtinProvider{
+		config: config,
+	}
+}
+
+func (p *builtinProvider) Capabilities() ([]string, error) {
+	return capabilities, nil
+}
+
+func (p *builtinProvider) Evaluate(cap string, conditionInfo interface{}) (lib.ProviderEvaluateResponse, error) {
+	response := lib.ProviderEvaluateResponse{}
+	switch cap {
+	case "file":
+		pattern, ok := conditionInfo.(string)
+		if !ok {
+			return response, fmt.Errorf("Could not parse provided file pattern as string: %v", conditionInfo)
+		}
+		matchingFiles, err := findFilesMatchingPattern(p.config.Location, pattern)
+		if err != nil {
+			return response, fmt.Errorf("Unable to find files using pattern `%s`: %v", pattern, err)
+		}
+
+		response.Passed = len(matchingFiles) == 0
+
+		for _, match := range matchingFiles {
+			response.ConditionHitContext = append(response.ConditionHitContext, map[string]string{
+				"filepath": match,
+			})
+		}
+		return response, nil
+	case "filecontent":
+		pattern, ok := conditionInfo.(string)
+		if !ok {
+			return response, fmt.Errorf("Could not parse provided regex pattern as string: %v", conditionInfo)
+		}
+		var outputBytes []byte
+		grep := exec.Command("grep", "-o", "-n", "-R", "-E", pattern, p.config.Location)
+		outputBytes, err := grep.Output()
+		if err != nil {
+			return response, fmt.Errorf("Could not run grep with provided pattern %+v", err)
+		}
+		matches := strings.Split(strings.TrimSpace(string(outputBytes)), "\n")
+		response.Passed = (len(matches) == 0)
+
+		for _, match := range matches {
+			//TODO(fabianvf): This will not work if there is a `:` in the filename, do we care?
+			pieces := strings.SplitN(match, ":", 3)
+			if len(pieces) != 3 {
+				//TODO(fabianvf): Just log or return?
+				return response, fmt.Errorf("Malformed response from grep, cannot parse %s with pattern {filepath}:{lineNumber}:{matchingText}", match)
+			}
+			response.ConditionHitContext = append(response.ConditionHitContext, map[string]string{
+				"filepath":     pieces[0],
+				"lineNumber":   pieces[1],
+				"matchingText": pieces[2],
+			})
```

Comment:
Is this more how you imagined `ConditionHitContext` being used?

Reply:
yes, but I think you are right, on the idea that some should be hardcoded

Reply:
we can follow up on that as part of pipelining I think

Reply:
or we could just adopt the data structures defined in https://github.com/konveyor/enhancements/pull/83

Reply:
Do you think that we are ready to do this now I am cool with breaking it out after if you are.